crankshaft-engine/src/service/runner.rs

```diff
@@ -65,17 +71,22 @@ impl Runner {
             backend,
             lock: Arc::new(Semaphore::new(max_tasks)),
             tasks: Default::default(),
+            generator: UniqueAlphanumeric::default_with_expected_generations(max_tasks),
         })
     }
 
     /// Submits a task to be executed by the backend.
-    pub fn submit(&self, task: Task) -> TaskHandle {
+    pub fn submit(&mut self, mut task: Task) -> TaskHandle {
         trace!(backend = ?self.backend, task = ?task);
 
         let (tx, rx) = tokio::sync::oneshot::channel();
         let backend = self.backend.clone();
         let lock = self.lock.clone();
 
+        if backend.default_name() == "docker" && task.name().is_none() {
+            task.set_name(self.generator.generate());
```

Comment:
This is the area I think can be improved on. Since the generator needs to be mutated it creates this whole upstream chain of method signatures that need to go from `&self` to `&mut self`